api/app/Models/User.php

```diff
@@ -619,23 +644,27 @@ public static function scopeEquity(Builder $query, ?array $equity): Builder
         return $query;
     }
 
-    public static function scopeGeneralSearch(Builder $query, ?string $search): Builder
+    public static function scopeGeneralSearch(Builder $query, ?array $searchTerms): Builder
     {
-        if ($search) {
-            $query->where(function ($query) use ($search) {
-                self::scopeName($query, $search);
-                $query->orWhere(function ($query) use ($search) {
-                    self::scopeEmail($query, $search);
-                });
-                $query->orWhere(function ($query) use ($search) {
-                    self::scopeTelephone($query, $search);
-                });
-            });
+        if ($searchTerms && is_array($searchTerms)) {
+            $searchResults = [];
+            foreach ($searchTerms as $searchTerm) {
+                // Use Scout's search method to perform the search and get results
+                $results = self::search($searchTerm)->usingPlainQuery()->get();
+                $searchResults = array_merge($searchResults, $results->all());
```

Comment:
Running a search on each term serially and merging the results seems not ideal.  

The first problem is that it will result in an OR search, right?  From the original issue: 
> Example 1: Process engineering, Forestry would retrieve candidates that have both "process engineering" (entire term / phrase) and "forestry" on their profile.

The second problem is that it doesn't allow the database to parallelize the multiple searches and instead forces them to go one-by-one.

Is there any way to send all the terms to the database at once?